packages/api/endpoints/s3credentials.js

```diff
@@ -0,0 +1,216 @@
+    userid: userId, // <- used by NGAP
+    policy,
+  });
+
+  return await lambda.invoke({
+    FunctionName: lambdaFunctionName,
+    Payload,
+  }).promise();
+}
+
+/**
+ * If DISABLE_S3_CREDENTIALS is not "true", returns undefined, otherwise, send a
+ * boom.ServerUnavailable to the caller, Exiting the request.
+ *
+ * @param {Object} res - express request object
+ * @returns {undefined} - when DISABLE_S3_CREDENTIALS is not 'true'
+ */
+function ensureEndpointEnabled(res) {
+  const disableS3Credentials = process.env.DISABLE_S3_CREDENTIALS;
+
+  if (disableS3Credentials && (disableS3Credentials.toLowerCase() === 'true')) {
+    return res.boom.serverUnavailable('S3 Credentials Endpoint has been disabled');
+  }
+  return undefined;
+}
+
+/**
+ * @returns {bool} whether or not the endpoint is configured to send ACL based
+ * credentials.
+ */
+function configuredForACLCredentials() {
+  if (process.env.CMR_ACL_BASED_CREDENTIALS && process.env.CMR_ACL_BASED_CREDENTIALS.toLowerCase() === 'true') {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Parses a "bucket/key/path" to return an array of ["bucket", "key/path"] where
+ * keypath is "/" if not specified.
+ *
+ * @param {string} bucketKeyPath
+ * @returns {Array<string>} Array [bucket, keypath]
+ */
+function parseBucketKey(bucketKeyPath) {
+  try {
+    const parts = bucketKeyPath.split('/');
+    const bucket = parts.shift();
+    const keypath = parts.join('/');
+    return { bucket, keypath: `/${keypath}` };
+  } catch (error) {
+    return {};
+  }
+}
+
+/**
+ * Reformats a list of buckets and bucket/keyspaths into the shape needed for
+ * calling the sts policy helper function. The desired payload is an object
+ * with 3 keys, 'accessmode', 'bucketlist' and 'pathlist'. 'bucketlist' and
+ * 'pathlist' are arrays of matching bucket and paths.
+ *
+ * Example:
+ * if the cmrAllowedBucketKeyList is:
+ * [ 'bucket1/somepath', 'bucket2']
+ * then the desired output would be the stringified object:
+ *
+ *  {
+ *   accessmode: 'Allow',
+ *   bucketlist: ['bucket1','bucket2'],
+ *   pathlist: ['/somepath', '/']
+ *  }
+ *
+ * @param {Array<string>} cmrAllowedBucketKeyList - earthdata login user name
+ * @returns {string} - stringified payload for policy helper function.
+ */
+function formatAllowedBucketKeys(cmrAllowedBucketKeyList) {
+  const bucketKeyPairList = cmrAllowedBucketKeyList.map(parseBucketKey);
+  const bucketlist = [];
+  const pathlist = [];
+
+  bucketKeyPairList.forEach((bucketKeyPair) => {
+    bucketlist.push(bucketKeyPair.bucket);
+    pathlist.push(bucketKeyPair.keypath);
+  });
+
+  return JSON.stringify({
+    accessmode: 'Allow',
+    bucketlist,
+    pathlist,
+  });
+}
+
+/**
+ *  Retrieve the sts session policy for a user when s3 credentials endpoint is
+ *  configured to use CMR ACLs. If the endpoint is not configured for CMR ACLs,
+ *  return undefined.
+ *
+ * @param {string} edlUser - earthdatalogin username
```

Comment:
Is this function EDL specific?  Also a couple of other functions mention about earthdata

Reply:
The getUserAccessibleBuckets uses that as CMR's user_id in the call to find out a users access to buckets https://cmr.earthdata.nasa.gov/access-control/site/docs/access-control/api.html#get-s3-buckets.  That is currently being derived from the token and it's assumed to be EDL.

Reply:
This is now behind a check for OAUTH_PROVIDER === 'earthdata' only

Reply:
So for current TEA distribution, OAUTH_PROVIDER is not set.  We should test if the CMR ACLs still works for TEA

Reply:
Fixed.